src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs

```diff
@@ -85,15 +85,22 @@ public class OutgoingReplicationHandler : IDisposable, IReportOutgoingReplicatio
 
         public OutgoingReplicationHandler(ReplicationLoader parent, DocumentDatabase database, ReplicationNode node, bool external, TcpConnectionInfo connectionInfo)
         {
-            _parent = parent;
-            _database = database;
-            Destination = node;
-            _external = external;
-            _log = LoggingSource.Instance.GetLogger<OutgoingReplicationHandler>(_database.Name);
-            _connectionInfo = connectionInfo;
-            _database.Changes.OnDocumentChange += OnDocumentChange;
-            _database.Changes.OnCounterChange += OnCounterChange;
-            _cts = CancellationTokenSource.CreateLinkedTokenSource(_database.DatabaseShutdown);
+            try
+            {
+                _parent = parent;
+                _database = database;
+                Destination = node;
+                _external = external;
+                _log = LoggingSource.Instance.GetLogger<OutgoingReplicationHandler>(_database.Name);
+                _connectionInfo = connectionInfo;
+                _database.Changes.OnDocumentChange += OnDocumentChange;
+                _database.Changes.OnCounterChange += OnCounterChange;
+                _cts = CancellationTokenSource.CreateLinkedTokenSource(_database.DatabaseShutdown);
+            }
+            catch (NullReferenceException)
+            {
+                // expected when DB is disposed
```

Comment:
I do not like this kind of coding. Please ping me and we will figure it out. If the race is between disposing and creating outgoingReplicationHandler, then we need to prevent both of those operations to run at the same time.

Reply:
Yes, there should never be something like that in our code.

Reply:
done